File: client/customer/src/main/java/com/reservation/customer/member/controller/MemberController.java
package com.reservation.customer.member.controller;

import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.bind.annotation.RestController;

import com.reservation.customer.member.controller.request.SignupRequest;
import com.reservation.customer.member.service.MemberService;
import com.reservation.support.response.ApiResponse;

import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;

@RestController
@Tag(name = "일반 회원 관리 API", description = "일반 회원 관리 API입니다.")
@RequiredArgsConstructor
public class MemberController {
	private final MemberService memberService;

	@PostMapping("/no-auth/member/signup")  //❗JWT auth 제외
	@Operation(summary = "회원 가입", description = "일반 고객 회원가입 API 입니다.")
	@ResponseStatus(HttpStatus.NO_CONTENT)
	public ApiResponse<Void> signup(@Valid @RequestBody SignupRequest request) {
		memberService.signup(request.email(), request.password(), request.phoneNumber());
		
		return ApiResponse.noContent();
	}
}

File: identity-service/src/main/kotlin/com/msa/identityservice/member/controller/MemberController.kt
package com.msa.identityservice.member.controller

import com.msa.identityservice.member.controller.request.MemberRegistrationRequest
import com.msa.identityservice.member.controller.response.MemberRegistrationResponse
import com.msa.identityservice.member.service.MemberService
import com.msa.identityservice.support.response.ApiResponse
import jakarta.validation.Valid
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/members")
class MemberController(
    private val memberService: MemberService
) {

    @PostMapping()
    @ResponseStatus(HttpStatus.CREATED)
    fun register(
        @Valid @RequestBody
        request: MemberRegistrationRequest
    ): ApiResponse<MemberRegistrationResponse> {
        val registerMemberDto = request.toRegisterMemberDto()
        val newMember = memberService.register(registerMemberDto)

        val response = MemberRegistrationResponse(
            id = newMember.id,
            email = newMember.email
        )

        return ApiResponse.create(
            message = "회원 가입에 성공했습니다.",
            data = response
        )
    }

}
